elytron/src/main/java/org/wildfly/extension/elytron/ModifiableRealmDecorator.java

```diff
@@ -503,31 +537,42 @@ protected void executeRuntimeStep(final OperationContext context, final ModelNod
         }
 
         private Password createPassword(final OperationContext parentContext, final String principalName, String passwordType, ModelNode passwordNode) throws OperationFailedException, NoSuchAlgorithmException, InvalidKeySpecException {
-            String password = SetPasswordHandler.Bcrypt.PASSWORD.resolveModelAttribute(parentContext, passwordNode).asString();
+            String password;
             final PasswordSpec passwordSpec;
             final String algorithm;
 
             if (passwordType.equals(ElytronDescriptionConstants.BCRYPT)) {
+                password = SetPasswordHandler.Bcrypt.PASSWORD.resolveModelAttribute(parentContext, passwordNode).asString();
                 byte[] salt = SetPasswordHandler.Bcrypt.SALT.resolveModelAttribute(parentContext, passwordNode).asBytes();
                 int iterationCount = SetPasswordHandler.Bcrypt.ITERATION_COUNT.resolveModelAttribute(parentContext, passwordNode).asInt();
                 passwordSpec = new EncryptablePasswordSpec(password.toCharArray(), new IteratedSaltedPasswordAlgorithmSpec(iterationCount, salt));
                 algorithm = SetPasswordHandler.Bcrypt.ALGORITHM.resolveModelAttribute(parentContext, passwordNode).asString();
             } else if (passwordType.equals(ElytronDescriptionConstants.CLEAR)) {
+                password = SetPasswordHandler.Bcrypt.PASSWORD.resolveModelAttribute(parentContext, passwordNode).asString();
```

Comment:
From the comments above, I'm not clear, Yeray, on whether the various PASSWORD attributes can all have the same settings or not. If they are all the same I think it's fine to declare the constant at the SetPasswordHandler level even though OTP doesn't use it. It's just a reasonable place for the constant. But if you don't like that it's fine to just have duplicate PASSWORD attributes for Bcrypt, Clear etc. Just use the related one in each of the if/else if blocks here so it's not confusing.